dvc/fs/dvc.py

```diff
@@ -292,7 +297,22 @@ def _get_fs_pair(
         Returns a pair of fss based on repo the path falls in, using prefix.
         """
         key = self._get_key_from_relative(path)
+        repo, fs, fs_path, dvc_fs, dvc_parts = self._get_fs_pair_2(key)
+        dvc_path = _get_dvc_path(dvc_fs, dvc_parts)
+        return repo, fs, fs_path, dvc_fs, dvc_path
 
+    def _get_fs_pair_2(
```

Comment:
The naming is kinda bad 😄

Reply:
Yeah, I know! I was hoping someone would have a better idea...

Reply:
@rlamy The `get_fs_pair` is also not great, it just stuck around from old times while changing the functionality. We should probably settle the naming for both. Your fresh look should be able to find better names for both (or feel free to reorganize those functions as you please to make new stuff make more sense).

Reply:
Yes, the naming issue is a sign that `_get_fs_pair_2()` doesn't make much sense...